wall_e/extensions/sfu.py

```diff
@@ -502,6 +506,111 @@ async def outline(self, ctx, *course):
         if e_obj is not False:
             await ctx.send(embed=e_obj, reference=ctx.message)
 
+    @app_commands.command(name="courses", description="Gets all offered courses")
+    @app_commands.describe(department="Specify the department to search. Examples: STAT, PHYS, MATH")
+    @app_commands.describe(level="Specifies the level of courses to filter for. Examples: 100, 200, 400")
+    @app_commands.describe(term="Specify the specific semester to search for. Examples: Spring, Summer, Fall")
+    @app_commands.describe(year="Specify the specific year to search for. Examples: 2020, 2024, 2025")
+    async def courses(self, interaction: discord.Interaction, department: str = "", level: int = 0,
+                  term: str = "registration", year: str = "registration"):
+        self.logger.info(
+            f'[SFU courses()] courses command detected from user {interaction.user} with arguments: '
+            f'department {department}, level {level}, term {term}, year {year}'
+        )
+        try:
+            await interaction.response.defer()
+        except discord.errors.NotFound:
```

Comment:
just an fyi, you don't need to do this try/except right out of the gate [if you notice, most of our command don't do this]. I placed this only on command that have actually complained about it as it proves they are used often enough to need it.